### PR TITLE
Add Feature-wise Linear Modulation layer

### PR DESCRIPTION
**Pull request by ethanjperez**
This PR adds an implementation for Feature-wise Linear Modulation (FiLM) layers, which scale and shift each feature (map) of a set of activations independently. FiLM is particularly useful for learning from multiple inputs/modalities and for incorporating external conditioning information.

This layer was introduced in "[FiLM: Visual Reasoning with a General Conditioning Layer](https://arxiv.org/abs/1709.07871)" and has since been used in follow-up work such as:
- "[Learning by Asking Questions](https://arxiv.org/abs/1712.01238)"
- "[Learning to Follow Language Instructions with Adversarial Reward Induction](https://arxiv.org/abs/1806.01946)"
- "[A dataset and architecture for visual reasoning with a working memory](https://arxiv.org/abs/1803.06092)"
- "[Guided Feature Transformation (GFT): A Neural
Language Grounding Module for Embodied Agents](https://arxiv.org/abs/1805.08329)"

If placed immediately after a batch norm layer with affine=false, this layer implements Conditional Normalization methods used in [visual question-answering](https://arxiv.org/abs/1707.00683), [image style transfer](https://arxiv.org/abs/1610.07629), and [speech recognition](https://arxiv.org/abs/1707.06065). Conditional Normalization was recently independently proposed as a [feature proposal](https://github.com/pytorch/pytorch/issues/8985) for PyTorch.

### CODE REVIEW

**Comment by SsnL** [comment]
Hey @ethanjperez , we feel that this is better suited for our [new contrib repo](https://github.com/pytorch/contrib). Since you submitted this PR before that repo existed, we will port it for you and let you know. :)

**Comment by ethanjperez** [comment]
@SsnL okay, great to hear! Thanks for porting it over and making the contrib repo
